services/web/project/app/views.py

```diff
@@ -63,29 +63,67 @@ def question(question_id):
         a = Answer(body, current_user.id, question.id)
         db.session.add(a)
         db.session.commit()
-        return redirect(url_for('views.question', question_id=question_id))
+        return redirect(url_for('views.question', question_id=question_id, user=current_user))
 
     # For each answer, add the creator as an attribute
     for a in question.answers:
         a.creator = User.query.get(a.userId)
-    return render_template('question.html', question=question, form=form)
+    return render_template('question.html', question=question, form=form, user=current_user)
 
 
 @views.route('/vote/<question_id>/<answer_id>/<value>', methods=['GET'])
 def vote(question_id, answer_id, value):
     answer_to_update = Answer.query.get(answer_id)
-    if int(value) == 1:
-        answer_to_update.numVotes += 1
-    else:
-        answer_to_update.numVotes -= 1
-    try:
+    num_votes_by_user = User.query.get(current_user.id).dailyVotes
+
+    if(num_votes_by_user >= 10):
+        flash('You have exceded 10 votes for the day. You will be allowed to vote again tomorrow at 6:30 am. Thank you',
+              category='error')
+        return redirect(request.referrer)
```

Comment:
You are redirecting here but your request is AJAX and the JS reloads the page as well. I would suggest either redirecting directly in your JS to the url you want and then let the Python method redirect or render a template. Otherwise don't redirect in the the back-end and send a message to your JS instead. The your ajax function can redirect there.